cli/src/commands/developer/deploy.rs

```diff
@@ -34,51 +37,66 @@ use snarkvm::{
 };
 
 use aleo_std::StorageMode;
-use anyhow::{Result, bail};
-use clap::Parser;
+use anyhow::Result;
+use clap::{Parser, builder::NonEmptyStringValueParser};
 use colored::Colorize;
 use snarkvm::prelude::{Address, ConsensusVersion};
 use std::{path::PathBuf, str::FromStr};
+use ureq::http::Uri;
 use zeroize::Zeroize;
 
+use anyhow::Context;
+
 /// Deploys an Aleo program.
 #[derive(Debug, Parser)]
+#[command(
+    group(clap::ArgGroup::new("mode").required(true).multiple(false)),
+    group(clap::ArgGroup::new("key").required(true).multiple(false))
+)]
 pub struct Deploy {
     /// The name of the program to deploy.
     program_id: String,
-    /// Specify the network to create a deployment for.
-    #[clap(default_value = "0", long = "network")]
-    pub network: u16,
     /// A path to a directory containing a manifest file. Defaults to the current working directory.
     #[clap(long)]
     path: Option<String>,
     /// The private key used to generate the deployment.
-    #[clap(short, long)]
-    private_key: String,
-    /// The endpoint to query node state from.
-    #[clap(short, long)]
-    query: String,
+    #[clap(short = 'p', long, group = "key", value_parser=NonEmptyStringValueParser::default())]
+    private_key: Option<String>,
+    /// Use a developer validator key tok generate the deployment.
+    #[clap(long, group = "key")]
+    dev_key: Option<u16>,
+    /// Specify the path to a file containing the account private key of the node
+    #[clap(long, group = "key", value_parser=NonEmptyStringValueParser::default())]
+    private_key_file: Option<String>,
+    /// The endpoint to query node state from and broadcast to (if set to broadcast).
+    #[clap(short, long, default_value=DEFAULT_ENDPOINT)]
+    endpoint: Uri,
     /// The priority fee in microcredits.
-    #[clap(long)]
+    #[clap(long, default_value_t = 0)]
     priority_fee: u64,
     /// The record to spend the fee from.
     #[clap(short, long)]
     record: Option<String>,
-    /// The endpoint used to broadcast the generated transaction.
-    #[clap(short, long, conflicts_with = "dry_run")]
-    broadcast: Option<String>,
+    /// Set the transaction to be broadcasted to the given endpoint.
+    #[clap(short, long, group = "mode")]
+    broadcast: bool,
```

Comment:
This will break usage for anyone relying on this flag today. The cleaner interface is not worth the many hours of developer time which this will cause. Please revert all backwards-incompatible changes of this PR.